src/app/components/modal/modal.service.ts

```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+
+@Injectable()
+export class ModalService {
+  private readonly $behaviorModal = new BehaviorSubject<boolean>(false);
+
+  public get listenerIsViewModal(): Observable<boolean> {
+    return this.$behaviorModal
```

Comment:
You should return an observable to give access to the source. So:

```suggestion
    return this.$behaviorModal.asObservable()
```
This block the possibility of emit or handle the state the `$behaviorModal`

Reply:
why an observable?, i think it's more efficent use a behavior subject than observable because in each new subscriber doesn't execute a new called

Reply:
and behavior subject only execute notifier to new subscriber went a value has been fired